### ISSUE TITLE
[PyTorch] [NEW API] No plot in weight decay

### ISSUE THREAD

**Issue by astonzhang** [d2l-ai/d2l-en#2026]
We need better plots to match our text discussions:

"We now run this code with lambd = 0, disabling weight decay. Note that we overfit badly, decreasing the training error but not the test error—a textbook case of overfitting."

`train_scratch(0)`

"Below, we run with substantial weight decay. Note that the training error increases but the test error decreases. This is precisely the effect we expect from regularization."

`train_scratch(3)`

http://preview.d2l.ai.s3-website-us-west-2.amazonaws.com/d2l-en/master/chapter_linear-networks/weight-decay.html
![Screen Shot 2022-01-26 at 4 48 06 PM](https://user-images.githubusercontent.com/22279212/151272514-65732e28-8bc5-4aef-b458-76af78ff4482.png)

![Screen Shot 2022-01-26 at 11 02 10 PM](https://user-images.githubusercontent.com/22279212/151308055-194f99ac-05f6-4728-a1fc-b41afb672eb8.png)

**Comment by astonzhang** [comment]
Fixed by https://github.com/d2l-ai/d2l-en/commit/b68076d927775a1d7dd434ce40c15239dc2ab8ac
